src/components/OnboardingManager.tsx

```diff
@@ -30,14 +33,17 @@ export default function OnboardingManager({
     // out of bounds redirect
     if (
       onboarding.progress < 0 ||
-      onboarding.progress >= onboarding.flow.length
+      onboarding.progress >= onboarding.flow.length ||
+      !profile?.userId ||
+      (profile?.profileReady && profile?.profileData)
```

Comment:
everything looks good! i just have some clarification questions 
when is `profile.profileReady` true? If it's only true when the entire profile is filled out, do we need to check `profileData` too? 
i'm assuming this line is to reroute when the user has already filled out onboarding?

Reply:
`profile.profileReady` is set to true after we fetch profile data from Supabase.
`profile.profileData` is what we get from Supabase.

Both are included to check:
1) we have the profile data from the user
2) the user has a profile (i.e. has already been onboarded)

If both are true, then we want to redirect the user because they have already finished onboarding (i.e. the goal of this PR).

Reply:
sounds good, i guess i'm just wondering if it's enough to check that we have profile data, i.e. just `profile?.profileData`. Doesn't this imply that the profile is ready?

Reply:
No, because `profileData` is undefined. Perhaps we can make an argument to check `undefined` vs `null`, but I don't think we have made a clear distinction yet (and it depends on the implementation of the underlying database). `profileReady` is an explicit boolean that we know we have set and can check to make sure the data is ready.